spec/models/certificate_expiry_reminder_spec.rb

```diff
@@ -0,0 +1,302 @@
+        template_id: "bbc34127-7fca-4d78-a95b-703da58e15ce",
+        personalisation: {
+          subject: "Your GOV.UK Verify certificates will expire on #{expected_expiry_date}",
+          team: cert.component.team.name,
+          no_of_certs: 1,
+          expire_on: expected_expiry_date,
+          certificates: ["Verify Service Provider (staging): encryption certificate - expires on #{cert.x509.not_after}"],
+          url: "http://www.test.com"
+        }
+      }
+
+      expect(a_request(:post, notify_endpoint).
+        with(body: expected_call.to_json)).
+        to have_been_made.once
+    end
+
+    it 'sends email if certificate is expiring in 7 days' do
+      stub_cognito_response(method: :list_users_in_group, payload: cognito_users)
+      stub_request(:post, notify_endpoint).to_return(body: '{}')
+
+      expires_in_days = 7
+      cert = create(:vsp_encryption_certificate, value: PKI.new.generate_encoded_cert(expires_in: expires_in_days.days))
+      certificate_expiry_reminder.run
+
+      expected_expiry_date = (Time.now + expires_in_days.days).strftime("%d %B %Y")
+      expected_call = {
+        email_address: email,
+        template_id: "bbc34127-7fca-4d78-a95b-703da58e15ce",
+        personalisation: {
+          subject: "Your GOV.UK Verify certificates will expire on #{expected_expiry_date}",
+          team: cert.component.team.name,
+          no_of_certs: 1,
+          expire_on: expected_expiry_date,
+          certificates: ["Verify Service Provider (staging): encryption certificate - expires on #{cert.x509.not_after}"],
+          url: "http://www.test.com"
+        }
+      }
+
+      expect(a_request(:post, notify_endpoint).
+        with(body: expected_call.to_json)).
+        to have_been_made.once
+    end
+
+    it 'sends emails if certificate is expiring in 7 days and team have many members' do
+      stub_cognito_response(method: :list_users_in_group, payload: many_cognito_users)
+      stub_request(:post, notify_endpoint).to_return(body: '{}')
+
+      expires_in_days = 7
+      cert = create(:vsp_encryption_certificate, value: PKI.new.generate_encoded_cert(expires_in: expires_in_days.days))
+      certificate_expiry_reminder.run
+
+      expected_expiry_date = (Time.now + expires_in_days.days).strftime("%d %B %Y")
+      expected_call = {
+        email_address: email,
+        template_id: "bbc34127-7fca-4d78-a95b-703da58e15ce",
+        personalisation: {
+          subject: "Your GOV.UK Verify certificates will expire on #{expected_expiry_date}",
+          team: cert.component.team.name,
+          no_of_certs: 1,
+          expire_on: expected_expiry_date,
+          certificates: ["Verify Service Provider (staging): encryption certificate - expires on #{cert.x509.not_after}"],
+          url: "http://www.test.com"
+        }
+      }
+
+      expect(a_request(:post, notify_endpoint).
+        with(body: expected_call.to_json)).
+        to have_been_made.times(3)
```

Comment:
I don't understand why this expects to send 3 times and not once

Reply:
it's because there are 3 users, so we expect to send 3 emails...

Reply:
Ok, not team members

Reply:
users = team members